src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -134,6 +134,10 @@ export class TextDiffEditor extends AbstractTextEditor<IDiffEditorViewState> imp
 			const control = assertIsDefined(this.diffEditorControl);
 			const resolvedDiffEditorModel = resolvedModel as TextDiffEditorModel;
 
+			if (resolvedDiffEditorModel) {
+				// Update the settings before creating the view model, as `waitForDiff` depends on the configuration
+				this.updateEditorConfiguration(resolvedDiffEditorModel.textDiffEditorModel!.modified.uri);
+			}
 			const vm = resolvedDiffEditorModel.textDiffEditorModel ? control.createViewModel(resolvedDiffEditorModel.textDiffEditorModel) : null;
 			await vm?.waitForDiff();
```

Comment:
What is the intent of `waitForDiff`? Was this added recently? Does this make opening the diff editor slower?

Reply:
Yes, I added this recently to reduce flickering significantly. With the collapse code feature this is basically a requirement, as we cannot show code just to collapse it some hundred milliseconds later.
But even before, the diff editor would show code and immediately scroll to the first diff and inserting alignment viewzones once the diff is computed. Maybe this should be limited to 2 seconds though.